### ISSUE TITLE
isEmpty errors on null

### ISSUE THREAD

**Issue by kwengontra** [remeda/remeda#775]
```
> R.isEmpty(null)
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Module.Wo (chunk-4DPIFY6Y.js:1:127)
    at <anonymous>:1:3
```

**Comment by eranhirsch** [comment]
isEmpty does not accept `null` as an input.

**Comment by kwengontra** [comment]
Could isEmpty be updated to accept null as an input? If not, what is the motivation?

**Comment by eranhirsch** [comment]
`null` is not a container and has no concept of being empty. Returning `false` for `isEmpty(null)` would be expected, but it's not the only semantically correct response. It's easy to do `isEmpty(x ?? [])` or even `x === null || isEmpty(x)` if that's the logic you need, but someone else might want to handle `null`s differently.

In Remeda, we prefer the input types be as strict as possible so that you get compile-time errors if a change upstream causes a mismatching type.

**Comment by J3m5** [comment]
> isEmpty does not accept `null` as an input.

This should be added to the docs IMO

**Comment by cjquines** [comment]
i think it's fair to add it to the docs, considering that `undefined` gets a mention. i'll reopen and rename the ticket

**Comment by scamden** [comment]
It's worth mentioning that on the other hand `R.isEmpty(undefined)` returns true.

**Comment by programming-with-ia** [comment]
Hey! 👋

I needed `isEmpty` to be fully type-safe and handle `null`, so I wrapped `remeda's` version like this (with a little help from GPT):

```

import { isEmpty as REmpty } from "remeda";

type IterableContainer<T = unknown> = ReadonlyArray<T> | readonly [];

// Overload: string | undefined | null
export function isEmpty<T extends string | undefined | null>(
    value: T,
): value is
    | ("" extends T ? "" : never)
    | (undefined extends T ? undefined : never)
    | (null extends T ? null : never);

// Overload: iterable containers (like array)
export function isEmpty(value: IterableContainer): value is [];

// Overload: plain object
export function isEmpty<T extends object>(
    value: T,
): value is Record<keyof T, never>;

// Implementation
export function isEmpty(value: unknown): boolean {
    if (value === null) return true;
    return REmpty(value as any);
}

```

**Comment by risalfajar** [comment]
> In Remeda, we prefer the input types be as strict as possible so that you get compile-time errors if a change upstream causes a mismatching type.

I agree with this, but there can be unexpected cases (like human errors), that can make the input be null. In this case, it will throw a runtime error, which is unpleasant.

What if a default error handling is added instead? Add a `try..catch` block that will return `true` on error.

**Comment by eranhirsch** [comment]
> but there can be unexpected cases (like human errors), that can make the input be null. In this case, it will throw a runtime error, which is unpleasant.

If a project has typing issues, it might throw in many other cases, too, not just when using Remeda utilities; that is not something Remeda is designed to address. Remeda is a **TypeScript** utility library that assumes the types in the project are sound.

e.g. 

```ts
// @ts-expect-error
const x: string = null;

// This would throw, it has nothing to do with Remeda
x.toLowerCase();
```
